### ISSUE TITLE
ER: New GHA to detect issues closed w/o PR or ignore

### ISSUE THREAD

**Issue by t-will-gillis** [hackforla/website#6291]
### Dependency
- [x] Post Board Migration

### Emergent Requirement - Problem

We want to make sure that if someone closes an issue that they are assigned to, without a PR attached or without being labelled `Ignore`, that the GHA:
- ??? Adds `ready for dev lead` label
- ??? moves the issue to "-----" column

### Issue you discovered this emergent requirement in
- #5749 

### Date discovered 
1/29/24 Dev Leads/ PM Meeting

### Did you have to do something temporarily
- [ ] YES
- [ ] NO

### Who was involved
@ExperimentsInHonesty @roslynwythe @JessicaLucindaCheng 

### What happens if this is not addressed
There is the possibility that we will lose track of an issue that has not been worked on or resolved yet.

### Recommended Action Items
- [X] Make a new issue
- [X] Discuss with team
- [ ] Let a Team Lead know

### Do the following
- [x] Review the issue draft below
- [x] identify if there are any changes you would need to make to the draft in order to work on the issue.
- [x] If there are changes needed, copy the draft into a comment and revise
- [x] once you think the draft is ready, add a `ready for merge team` label and put this ER in the Questions column
- [x] once approved, 
   - [x] create the issue
   - [x] link it in a comment below 
   - [x] add `ready for product` label to this issue and the `ready for prioritization` label to the issue you create.

Create issuer

### Title
```
Create GitHub Actions: Detect issues closed w/o PR or ignore
```

### Body
```
### Action Items
- [ ] Create a label: non-PR contribution
   - this label is to be used for:
      - wiki additions/revisions
      - audit spreadsheet contributions
      - Issue Making ERs and Epics
      - etc.
- [ ] Write a GitHub Action with the following features
   - [ ] The automation should reopen issues that are closed without:
      - PR
      - ignore label
      - non-PR contribution label
   - [ ] The automation should add the label: `ready for product`
   - [ ] The automation should leave a comment on the issue with this text
      ```
      This issue was reopened because it did not have a
      - PR
      - ignore label
      - non-PR contribution label
      ```
   - [ ] The automation should move the issue to the Questions / In Review column

### Resources/Instructions
Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)

Note that you might want to do something outside the scope of the above psudo-code. If so, be sure to leave comments in your PR or this issue that justifies your reasoning. If you feel you need guidance, be sure to reach out! We cannot foresee whether this issue is solvable, or what hard decisions have to be made, but we would love to hear and help you!

#### Additional resources:
[Events that trigger workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)
[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
[GitHub GraphQL](https://docs.github.com/en/graphql)
```

**Comment by HackforLABot** [comment]
Hi @dvernon5, thank you for taking up this issue! Hfla appreciates you :)

Do let fellow developers know about your:-
i. **Availability:** (When are you available to work on the issue/answer questions other programmers might have about your issue?)
ii. **ETA:** (When do you expect this issue to be completed?)

You're awesome!

P.S. - You may not take up another issue until this issue gets merged (or closed). Thanks again :)

**Comment by dvernon5** [comment]
@HackforLABot 

Availability: Monday - Thursday 12pm - 6pm | Friday 12pm - 4pm
ETA: 7/12/2025

**Comment by dvernon5** [comment]
### Action Items
- [ ] Create a label: non-PR contribution
   - this label is to be used for:
      - wiki additions/revisions
      - audit spreadsheet contributions
      - Issue Making ERs and Epics
      - etc.
- [ ] Write a GitHub Action with the following features
   - [ ] The automation should reopen issues that are closed without:
      - PR
      - ignore label
      - non-PR contribution label
   - [ ] The automation should add the label: `ready for dev lead`
   - [ ] The automation should leave a comment on the issue with this text
      ```
      This issue was reopened because it did not have a
      - PR
      - ignore label
      - non-PR contribution label
      ```
   - [ ] The automation should move the issue to the Questions / In Review column

### Resources/Instructions
Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)

Note that you might want to do something outside the scope of the above psudo-code. If so, be sure to leave comments in your PR or this issue that justifies your reasoning. If you feel you need guidance, be sure to reach out! We cannot foresee whether this issue is solvable, or what hard decisions have to be made, but we would love to hear and help you!

#### Additional resources:
[Events that trigger workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)
[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
[GitHub GraphQL](https://docs.github.com/en/graphql)


I think the draft looks great! If I'm understanding the task correctly, the only change that I made regarding the potential issue was updating `- [ ] The automation should add the label: `ready for product`` to `- [ ] The automation should add the label: `ready for dev lead`` 

Please let me know if the modifications I made are accurate.

**Comment by dvernon5** [comment]
Create GitHub Actions: Detect issues closed w/o PR or ignore: https://github.com/hackforla/website/issues/8139

**Comment by t-will-gillis** [comment]
Closing this ER - #8139 created
